lib/threads.cpp

```diff
@@ -1111,6 +1157,48 @@ void communication_thread::execute()
             goto comm_terminate;
          }
       }
+      else if (ident == "axis_numberformat")
+      {
+         char c;
+         string formatting;
+
+         *is[0] >> ws >> c; // read the opening char
+         getline(*is[0], formatting, c); // read formatting string & use c for termination
+
+         // all processors sent the command
+         for (size_t i = 1; i < is.size(); i++)
+         {
+            *is[i] >> ws >> ident; // 'axis_numberformat'
+            *is[i] >> ws >> c;
+            getline(*is[i], ident, c);
+         }
+
+         if (glvis_command->AxisNumberFormat(formatting))
+         {
+            goto comm_terminate;
+         }
+      }
+      else if (ident == "colorbar_numberformat")
+      {
+         char c;
+         string formatting;
+
+         *is[0] >> ws >> c; // read the opening char
+         getline(*is[0], formatting, c); // read formatting string & use c for termination
```

Comment:
Is it necessary to enclose the format in opening/closing characters? We do not expect spaces in the format string (unlike in the caption, which uses `getline`), so just a whitespace could end the string.

Reply:
The format string can have a space, e.g. "% g" is valid format string.

Reply:
Oh, that is true, I have never used that option for padding the output, but nothing very useful, I think. Up to @justinlaughlin , it is just a suggestion...

Reply:
Didn't know that about the space, also apparently `'` can be used as part of the format string e.g. `%'2.3f`. We could use the regex from https://github.com/GLVis/glvis/pull/288/commits/f91f3e47217ee5700d3ce833767d9e22b27ef9a0 to not have delimiters. Personally I'm partial to keeping this as-is since it is simple + similar to neighboring code.

Reply:
I do not see the apostrophe [here](https://en.cppreference.com/w/c/io/fprintf) 🤔 I would drop it, making it easier with loading the string here :wink:.

Reply:
oops 😅 no idea where that came from.